### DISCUSSION TITLE
guava version updates are deadlocked due to GitHub vulnerability alert

### DISCUSSION THREAD

**Discussion by Shegox** [renovatebot/renovate#22909] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

tested as well with 35.133.3 against GitHub Enterprise Server

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I noticed that for the [`com.google.guava:guava` dependency](https://mvnrepository.com/artifact/com.google.guava/guava) a (wrong) GitHub vulnerability alert exist and this causes Renovate to no longer propose **any** updates to this dependency (e.g. an update from `32.0.0-jre` -> `32.0.1-jre`) creating an effective deadlock for this dependency.

The vulnerability alert is for releases below `<32.0.0` (https://github.com/advisories/GHSA-7g45-4rm6-3mm3).

The library itself however uses versions like `32.0.0-jre` and `32.0.0-android`  (with a compatibility suffix), which is not supported by the default maven versioning. As the vulnerability alert now specifies as fixed versions `32.0.0` Renovate uses this version to construct the `allowedVersions` like this in a packageRule: 
```
      "matchCurrentVersion": "32.0.0-jre",
      "allowedVersions": "32.0.0",
```

The problem seems to be two fold: 
- GitHub detects `32.0.0-jre` as within the affected versions (`>= 1.0, < 32.0.0`) and triggers a vulnerability alert
- Renovate uses the `allowedVersions` to restrict the allowed Versions to a single version (`32.0.0`) and then dead-locks the dependency from further updates as this exact version doesn't (technically) exist.

---

I tried as well specifying a [custom regex versioning for guava](https://docs.renovatebot.com/modules/versioning/#regular-expression-versioning), but that didn't change the observed behavior.
Manually dismissing the GitHub vulnerability alert restores the desired behavior and Renovate will propose an update from `32.0.0-jre` to `32.0.1-jre`

Reproduction repository: https://github.com/Shegox/guava-renovate/pull/1

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: GitHub vulnerability details
{
  "alerts": {
    "maven/com.google.guava:guava": {
      ">= 1.0, < 32.0.0": "32.0.0"
    }
  }
}

DEBUG: alert package rules
{
  "alertPackageRules": [
    {
      "matchDatasources": [
        "maven"
      ],
      "matchPackageNames": [
        "com.google.guava:guava"
      ],
      "matchCurrentVersion": "32.0.0-jre",
      "matchFiles": [
        "pom.xml"
      ],
      "allowedVersions": "32.0.0",
      "prBodyNotes": [
        "### GitHub Vulnerability Alerts",
        "#### [CVE-2023-2976](https://nvd.nist.gov/vuln/detail/CVE-2023-2976)\n\nUse of Java's default temporary directory for file creation in `FileBackedOutputStream` in Google Guava versions 1.0 to 31.1 on Unix systems and Android Ice Cream Sandwich allows other users and apps on the machine with access to the default Java temporary directory to be able to access the files created by the class.\n\nEven though the security vulnerability is fixed in version 32.0.0, maintainers recommend using version 32.0.1 as version 32.0.0 breaks some functionality under Windows.\n\n"
      ],
      "isVulnerabilityAlert": true,
      "force": {
        "groupName": null,
        "schedule": [],
        "dependencyDashboardApproval": false,
        "minimumReleaseAge": null,
        "rangeStrategy": "update-lockfile",
        "commitMessageSuffix": "[SECURITY]",
        "branchTopic": "{{{datasource}}}-{{{depName}}}-vulnerability",
        "prCreation": "immediate"
      }
    }
  ]
}
....
DEBUG: Looking up com.google.guava:guava in repository https://repo.maven.apache.org/maven2/
DEBUG: Found 112 new releases for com.google.guava:guava in repository https://repo.maven.apache.org/maven2/
...
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "maven": [
      {
        "datasource": "maven",
        "packageFile": "pom.xml",
        "deps": [
          {
            "datasource": "maven",
            "depName": "com.google.guava:guava",
            "currentValue": "32.0.0-jre",
            "fileReplacePosition": 422,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2"
            ],
            "depType": "compile",
            "updates": [],
            "packageName": "com.google.guava:guava",
            "warnings": [],
            "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$",
            "sourceUrl": "https://github.com/google/guava",
            "registryUrl": "https://repo.maven.apache.org/maven2",
            "currentVersion": "32.0.0-jre",
            "fixedVersion": "32.0.0-jre"
          }
        ]
      }
    ]
  }
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Dismissing an incorrect vulnerability alert seems like the best solution? It's not like there is any *downside* to that?

**Reply by redehnroV** [reply]
You have the same issue when it is a correct issue, the issue leads to the behaviour, that you don't get the automatic update for a dependency, where you rely on the renovate tool, to update all dependencies.

**Reply by rarkins** [reply]
It's not the same
